server/src/main/java/org/opensearch/cluster/coordination/JoinTaskExecutor.java

```diff
@@ -140,6 +143,11 @@ public ClusterTasksResult<Task> execute(ClusterState currentState, List<Task> jo
         ClusterState.Builder newState;
 
         if (joiningNodes.size() == 1 && joiningNodes.get(0).isFinishElectionTask()) {
+            DiscoveryNode joiningNode = joiningNodes.get(0).node();
+            if (isRemoteStoreNode(joiningNode)) {
```

Comment:
nit - we can simplify the if else logic here by determining remote status of new node and existing node upfront and making this if else a  simple condition. 

`
boolean newNodeRemoteStatus = isRemoteStoreNode(joiningNode)
boolean existingNodeRemoteStatus = isRemoteStoreNode(existingNodes.get(0))
boolean sameAttributes = compareNodeAttributes(joiningNode, existingNode) // let this return boolean
if ((newNodeRemoteStatus != existingNodeRemoteStatus) || (newNodeRemoteStatus && sameAttributes == false)) {
   // throw..
}
`